File: backend/src/main/java/com/galinho/backend/model/Usuarios/Cliente.java
package com.galinho.backend.model.Usuarios;

import com.galinho.backend.utils.Role;
import jakarta.persistence.Entity;
import jakarta.persistence.PrimaryKeyJoinColumn;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@ToString
@Entity
@PrimaryKeyJoinColumn(name = "idUsuario")
public class Cliente extends Usuario {

    private String endereco;

    public Cliente(String email,
                   String senha,
                   String nome,
                   String cpf,
                   String telefone,
                   String endereco) {
        this.email = email;
        this.senha = senha;
        this.nome = nome;
        this.cpf = cpf;
        this.telefone = telefone;
        this.dataCadastro = LocalDate.now();
        this.role = Role.CLIENTE;
        this.endereco = endereco;
    }
}

File: frontend/src/interfaces/Cliente.ts
import type { List } from "lodash";
import type { Veiculo } from "./Veiculo";

export interface Cliente{
    id?: number;
    nome: string;
    email: string;
    senha: string;
    cpf: string;
    telefone: string;
    dataCadastro?: Date;
    endereco: string;
    veiculos?: List<Veiculo>;
    role?: string;
}